pelican/contents.py

```diff
@@ -520,12 +520,17 @@ def refresh_metadata_intersite_links(self):
 
         # _summary is an internal variable that some plugins may be writing to,
         # so ensure changes to it are picked up
-        if (
-            "summary" in self.settings["FORMATTED_FIELDS"]
-            and "summary" in self.metadata
-        ):
-            self._summary = self._update_content(self._summary, self.get_siteurl())
-            self.metadata["summary"] = self._summary
+        if "summary" in self.settings["FORMATTED_FIELDS"]:
+            if hasattr(self, "_summary"):
+                self.metadata["summary"] = self._summary
+
+            if "summary" in self.metadata:
+                self.metadata["summary"] = self._update_content(
+                    self.metadata["summary"], self.get_siteurl()
+                )
+
+            if hasattr(self, "_summary") and "summary" in self.metadata:
+                self._summary = self.metadata["summary"]
```

Comment:
To be precise, I'm talking about the last `if` condition here, specifically `hasattr(self, "_summary")`. If you compare to the previous version, this `if` should not be here and their contents should be combined to ensure that `._summary` and `metadata['summary']` are synced:

```python
            if "summary" in self.metadata:
                self.metadata["summary"] = self._update_content(
                    self.metadata["summary"], self.get_siteurl()
                )
                self._summary = self.metadata["summary"]